server/api/db/sqldb/db.py

```diff
@@ -896,9 +896,19 @@ def tag_artifacts(
         artifacts,
         project: str,
     ):
+        # to avoid multiple runs trying to tag the same artifacts simultaneously,
```

Comment:
What happens if some of the artifacts are locked? I assume it will wait until it can lock all rows but does it mean that the other rows are locked until the lock we wait for is released? I would assume that mysql has a good mechanism for it (FCFS?) but we should verify it